src/shared/hooks/reflect.ts

```diff
@@ -0,0 +1,108 @@
+import { useEffect } from "react"
+import {
+  selectDisplayedRealmId,
+  selectStakingRealmId,
+  selectWalletName,
+  useDappSelector,
+} from "redux-state"
+import { usePresence, useSubscribe } from "@rocicorp/reflect/react"
+import { getClientState, reflectInstance } from "shared/utils"
+import { getRealmMapData } from "shared/constants"
+import { ReflectClient } from "shared/types"
+
+export function useReflect() {
+  const name = useDappSelector(selectWalletName)
+  const stakingRealmId = useDappSelector(selectStakingRealmId)
+
+  const avatar = stakingRealmId
+    ? getRealmMapData(stakingRealmId).partnerIcons.default
```

Comment:
Let's save `getRealmMapData(stakingRealmId)` once and then destructure or provide default values